### ISSUE TITLE
Equivalent to `nixos-rebuild boot` ?

### ISSUE THREAD

**Issue by danbst** [NixOS/nixops#643]
What is a nixops quivalent to `nixos-rebuild boot`? I have an instance with backend `none` and would like to perform upgrade 16.09 -> 17.03 with least pain.

`nixops deploy` can do `--force-reboot` only for machines in cloud, can do `--build-only`, `test-only`, `--dry-activate`, but can't activate deployment only after reboot.

**Comment by domenkozar** [comment]
Basically a duplicate of https://github.com/NixOS/nixops/issues/643

**Comment by danbst** [comment]
I'd say, a dual )

**Comment by vincentbernat** [comment]
From my understanding, nixops doesn't keep generations on the server, but on the host running nixops. So, it's not easy to get `nixos-rebuild boot`.

**Comment by wizeman** [comment]
I've been using the following one-line patch for a while and it works fine for me:
```diff
diff --git a/nixops/deployment.py b/nixops/deployment.py
index 5a1b423b..4248284f 100644
--- a/nixops/deployment.py
+++ b/nixops/deployment.py
@@ -1031,7 +1031,7 @@ class Deployment:
                     )
 
                 if res == 100 or force_reboot or m.state == m.RESCUE:
-                    if not allow_reboot and not force_reboot:
+                    if not allow_reboot:
                         raise Exception(
                             "the new configuration requires a "
                             "reboot of '{}' to take effect (hint: use "
```

After applying this patch, when I want the equivalent of `nixos-rebuild boot` I will do `nixops deploy --force-reboot` (it will give an error saying that it can't reboot because `--allow-reboot` was not provided, but it's harmless). If I want the host to actually reboot, I will do `nixops deploy --force-reboot --allow-reboot`.

Obviously this simple patch is not ready to be upstreamed but I don't imagine it would be very difficult to make it more user-friendly and backwards-compatible.
